packages/headless/src/features/search/search-actions-loader.ts

```diff
@@ -68,5 +83,6 @@ export function loadSearchActions(engine: SearchEngine): SearchActionCreators {
   return {
     executeSearch,
     fetchMoreResults,
+    prepareForSearchWithQuery,
```

Comment:
I'd prefer if we kept this action internal to start. It's a higher-level building block that bakes in a UX Coveo believes is correct (deselecting all facets). Exposed actions are typically atomic operations that modify one thing, so this action is kind of different.

In the worst-case, if someone really needed it, they can replicate it since all the individual actions are exposed.